roles/docker/vars/ubuntu.yml

```diff
@@ -8,9 +8,11 @@ docker_versioned_pkg:
   '1.12': docker-engine=1.12.6-0~ubuntu-{{ ansible_distribution_release|lower }}
   '1.13': docker-engine=1.13.1-0~ubuntu-{{ ansible_distribution_release|lower }}
   '17.03': docker-ce=17.03.2~ce-0~ubuntu-{{ ansible_distribution_release|lower }}
+  '17.12': docker-ce=17.03.2~ce-0~ubuntu-{{ ansible_distribution_release|lower }}
```

Comment:
mmm it looks like 17.12 should match docker-ce=17.12 instead of 17.03. Somehow this does not make the cluster disfunctional, but it's not nice.
On the 18.03 question, docker-ce version also needs to match the available nvidia-runtime version. If 18.03 runimte for nvidia is not (yet) available, you should not run docker 18.03 (at least not with a gpu setup)